pkg/loop/internal/runner/runner.go

```diff
@@ -0,0 +1,227 @@
+package runner
+
+import (
+	"context"
+	"debug/elf"
+	"debug/macho"
+	"debug/pe"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/user"
+	"runtime"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin/runner"
+)
+
+type logMessage struct {
+	Level     string    `json:"@level"`
+	Timestamp time.Time `json:"@timestamp"`
+	Logger    string    `json:"logger"`
+	Caller    string    `json:"caller"`
+	Message   string    `json:"@message"`
+}
+
+var _ runner.Runner = (*Runner)(nil)
+
+type Runner struct {
+	logger  hclog.Logger
+	cmd     *exec.Cmd
+	cmdPath string
+	cmdArgs []string
+	cmdEnv  []string
+
+	stdout io.ReadCloser
+	stderr io.ReadCloser
+
+	pid int
+}
+
+var _ io.ReadCloser = (*stderrWrapper)(nil)
+
+type stderrWrapper struct {
+	r    io.ReadCloser
+	lggr hclog.Logger
+}
+
+func NewIOWrapper(l hclog.Logger, r io.ReadCloser) *stderrWrapper {
+	return &stderrWrapper{
+		r:    r,
+		lggr: l,
+	}
+}
+func (i *stderrWrapper) logIfPanic(m []byte) {
+	//Normal logs are pushed in here as well, to distinguish them from panics we try to unmarshall them as json
+	l := &logMessage{}
+	err := json.Unmarshal(m[:len(m)-1], l) //Remove trailing newline
+	if err != nil {
+		i.lggr.Error(string(m))
+	}
+}
+
+func (i *stderrWrapper) Read(p []byte) (n int, err error) {
+	n, err = i.r.Read(p)
+	if err != nil {
+		return 0, nil
+	}
+	i.logIfPanic(p[:n])
+	return
+}
+
+func (i *stderrWrapper) Close() error {
+	i.r.Close()
+	return nil
+}
+
+func NewRunnerFunc(path string, args, env []string) func(l hclog.Logger, cmd *exec.Cmd, tmpDir string) (runner.Runner, error) {
+	cr := &Runner{
+		cmdPath: path,
+		cmdArgs: args,
+		cmdEnv:  env,
+	}
+	return cr.NewCmdRunner
+}
+
+func (c *Runner) NewCmdRunner(logger hclog.Logger, cmd *exec.Cmd, tmpDir string) (runner.Runner, error) {
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return nil, err
+	}
+
+	c.logger = logger
+	c.cmd = cmd
+	c.cmd.Path = c.cmdPath
+	c.cmd.Args = c.cmdArgs
+	c.cmd.Env = append(c.cmd.Env, c.cmdEnv...)
+	c.stdout = stdout
+	c.stderr = NewIOWrapper(logger, stderr)
```

Comment:
i don't think we need a custom runner. instead, i think we can specific the stderr writer in the client config.

look at https://github.com/smartcontractkit/chainlink-common/blob/d1a00d5221f3328e7b74ccb17f891ba3bd227aa3/pkg/loop/plugin_relayer.go#L60

for example. and https://github.com/hashicorp/go-plugin/blob/8d2aaa458971cba97c3bfec1b0380322e024b514/client.go#L197

that client config lets us declare a stderr handle. you could add a 'panicWriter` config wrapper 

~ NewPanicWriterConfig(in *clientConfig) *clientConfig {

return &ClientConfig {
...
Stderr: NewPanicIOWrapper(in.logger, in.stderr)
}

and use this `panicWriter` everywhere to achieve the same thing that your custom runner is doing

Reply:
another thought is that you might be able to intercept the panic here, in our hclogger wrapper https://github.com/smartcontractkit/chainlink-common/blob/d1a00d5221f3328e7b74ccb17f891ba3bd227aa3/pkg/loop/logger.go#L69

if you know that `panics` are written a debug level, then in principle you might be able to put a parser in that switch and re-route then to Error level.